docs/tutorial/go.rst

```diff
@@ -0,0 +1,351 @@
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+
+Update Go application
+========================================
+
+As a final step, let's update our application. For example,
+we want to add a new ``/time`` endpoint which returns the current time.
+
+Start by opening the ``main.go`` file in a text editor and update the code to
+look like the following:
+
+.. literalinclude:: code/go/main.go.time
+    :language: go
+
+Since we are creating a new version of the application, **open the**
+``rockcraft.yaml`` file and change the ``version`` (e.g. to ``0.2``).
+
+.. note::
+
+    ``rockcraft pack`` will create a new image with the updated code even if you
+    don't change the version. It is recommended to change the version whenever
+    you make changes to the application in the image.
+
+Pack and run the rock using similar commands as before:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:docker-run-update]
+    :end-before: [docs:docker-run-update-end]
+    :dedent: 2
+
+.. note::
+
+    Note that the resulting ``.rock`` file will now be named differently, as
+    its new version will be part of the filename.
+
+Finally, use ``curl`` to send a request to the ``/time`` endpoint:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:curl-time]
+    :end-before: [docs:curl-time-end]
+    :dedent: 2
+
+The updated application should respond with the current date and time (e.g.
+``2024-06-21 09:47:56``).
+
+.. note::
+
+    If you are not getting the current date and time from the ``/time``
+    endpoint, check the :ref:`troubleshooting-go` steps below.
+
+
+Cleanup
+~~~~~~~
+
+We can now stop the container and remove the corresponding image:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker-updated]
+    :end-before: [docs:stop-docker-updated-end]
+    :dedent: 2
+
+
+Reset your environment
+======================
+
+We've reached the end of this tutorial.
+
+If you'd like to reset your working environment, you can simply run the
+following:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:cleanup]
+    :end-before: [docs:cleanup-end]
+    :dedent: 2
+
+.. collapse:: If using Multipass...
+
+    If you created an instance using Multipass, you can also clean it up.
+    Start by exiting it:
+
+    .. code-block:: bash
+
+        exit
+
+    And then you can proceed with its deletion:
+
+    .. code-block:: bash
+
+        multipass delete rock-dev
+        multipass purge
+
+----
+
+.. _troubleshooting-go:
+
+Troubleshooting
+===============
+
+**Application updates not taking effect?**
+
+Upon changing your Go application and re-packing the rock, if you believe
+your changes are not taking effect, try running ``rockcraft clean`` and pack
+the rock again with ``rockcraft pack``.
+
```

Comment:
Maybe in the end, we can add something like "what's next", "where to go from here", to add more links for the users, like links to explanations and how-tos.

Reply:
Sounds reasonable, we should do that for the rest of the tutorials as well

Reply:
I have added a section "next steps" (following the juju docs nomenclature). Let's discuss what to put in here for this tutorial and do it for the rest of the tutorials in another PR.